### PR TITLE
Down integrate internal changes to GitHub

### PR DESCRIPTION
**Pull request by rafi-kamal**

### CODE REVIEW

File: Protobuf-C++.podspec [hunk 1]
```diff
@@ -1,42 +0,0 @@
-Pod::Spec.new do |s|
```
Comment (TeBoring) [inline Protobuf-C++.podspec]: This file should not be deleted.
I guess we need to add this to some internal list.

File: build_files_updated_unittest.sh [hunk 2]
```diff
@@ -1,62 +0,0 @@
-#!/bin/bash
```
Comment (TeBoring) [inline build_files_updated_unittest.sh]: This should not be deleted

File: conformance/binary_json_conformance_suite.cc [hunk 3]
```diff
@@ -922,107 +895,106 @@ void BinaryAndJsonConformanceSuite::RunSuiteImpl() {
...
+                           {delim("\xfb"), delim("\xfb")},
+                           {delim(""), delim("")},
+                       });
   TestValidDataForType(FieldDescriptor::TYPE_ENUM, {
-    {varint(0), varint(0)},
-    {varint(1), varint(1)},
-    {varint(2), varint(2)},
-    {varint(-1), varint(-1)},
-  });
+                                                       {varint(0), varint(0)},
```
Comment (TeBoring) [inline conformance/binary_json_conformance_suite.cc]: why so many spaces?

File: conformance/failure_list_js.txt [hunk 4]
```diff
@@ -1,113 +1,13 @@
-Recommended.Proto2.ProtobufInput.ValidDataRepeated.BOOL.PackedInput.DefaultOutput.ProtobufOutput
```
Comment (TeBoring) [inline conformance/failure_list_js.txt]: I think you should pick the github change for this file

File: update_compatibility_version.py [hunk 5]
```diff
@@ -1,57 +0,0 @@
-#!/usr/bin/env python
```
Comment (TeBoring) [inline update_compatibility_version.py]: You also need to keep this

File: conformance/binary_json_conformance_suite.cc (hunk 3 above)
Comment (rafi-kamal) [reply conformance/binary_json_conformance_suite.cc]: That also looks odd to me. But unfortunately even if I remove the spaces, g4 fix brings them back.

File: conformance/failure_list_js.txt (hunk 4 above)
Comment (rafi-kamal) [reply conformance/failure_list_js.txt]: Done

File: src/Makefile.am [hunk 6]
```diff
@@ -213,6 +213,7 @@ libprotobuf_lite_la_SOURCES =                                  \
   google/protobuf/repeated_field.cc                            \
   google/protobuf/wire_format_lite.cc                          \
   google/protobuf/io/coded_stream.cc                           \
+  google/protobuf/io/coded_stream_inl.h                        \
```
Comment (TeBoring) [inline src/Makefile.am]: This line needs to be removed. Please double check it's removed when you do up-integration.

Comment (rafi-kamal) [reply src/Makefile.am]: Done

**Pull request closed without merging**